src/main/java/uk/gov/justice/digital/delius/service/OffenderService.java

```diff
@@ -142,4 +143,27 @@ public Long getOffenderCount() {
                 offender -> offenderTransformer.responsibleOfficersOf(offender, current));
 
     }
+
+    @Transactional(readOnly = true)
+    public Optional<List<CommunityOrPrisonOffenderManager>> getAllOffenderManagersForNomsNumber(String nomsNumber) {
+        return offenderRepository.findByNomsNumber(nomsNumber).map(
+                offender -> combine(
+                        offender.getOffenderManagers()
+                                .stream()
+                                .filter(uk.gov.justice.digital.delius.jpa.standard.entity.OffenderManager::isActive)
+                                .map(offenderManagerTransformer::offenderManagerOf)
+                                .collect(Collectors.toList()),
+                        offender.getPrisonOffenderManagers()
+                                .stream()
+                                .filter(uk.gov.justice.digital.delius.jpa.standard.entity.PrisonOffenderManager::isActive)
+                                .map(offenderManagerTransformer::offenderManagerOf)
+                                .collect(Collectors.toList())
+                ) );
+    }
+
+     private static <T> List<T> combine(List<T> first, List<T> second) {
```

Comment:
couldn't find anything in JDK that merged two lists in a functional way

Reply:
agreed, that's the way I normally do it